### ISSUE TITLE
Error: undefined method `/' for nil:NilClass

### ISSUE THREAD

**Issue by xrisk** [Homebrew/homebrew#50185]
Hi, I'm getting an error when trying to run any of the brew commands other than `update`.

```
➜  ~ brew upgrade
Error: undefined method `/' for nil:NilClass
Please report this bug:
    https://git.io/brew-troubleshooting
/usr/local/Library/Homebrew/os/mac/xcode.rb:82:in `uncached_version'
/usr/local/Library/Homebrew/os/mac/xcode.rb:71:in `version'
/usr/local/Library/Homebrew/diagnostic.rb:419:in `check_for_osx_gcc_installer'
/usr/local/Library/Homebrew/cmd/install.rb:169:in `block in check_xcode'
/usr/local/Library/Homebrew/cmd/install.rb:168:in `each'
/usr/local/Library/Homebrew/cmd/install.rb:168:in `check_xcode'
/usr/local/Library/Homebrew/cmd/install.rb:194:in `perform_preinstall_checks'
/usr/local/Library/Homebrew/cmd/upgrade.rb:8:in `upgrade'
/usr/local/Library/brew.rb:83:in `<main>'
```

The same goes for `brew doctor`:

```
➜  ~ brew doctor
Error: undefined method `/' for nil:NilClass
Please report this bug:
    https://git.io/brew-troubleshooting
/usr/local/Library/Homebrew/os/mac/xcode.rb:82:in `uncached_version'
/usr/local/Library/Homebrew/os/mac/xcode.rb:71:in `version'
/usr/local/Library/Homebrew/diagnostic.rb:419:in `check_for_osx_gcc_installer'
/usr/local/Library/Homebrew/cmd/doctor.rb:34:in `block in doctor'
/usr/local/Library/Homebrew/cmd/doctor.rb:27:in `each'
/usr/local/Library/Homebrew/cmd/doctor.rb:27:in `doctor'
/usr/local/Library/brew.rb:83:in `<main>'
```

I've tried the instructions [here](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Common-Issues.md#brew-upgrade-errors-out) but the first `cd` points to a path that doesn't exist on my computer. I've also tried `sudo chown -R $(whoami) /usr/local` but that didn't help either.

Thanks.

**Comment by xrisk** [comment]
`brew config`:

```
➜  ~ brew config
HOMEBREW_VERSION: 0.9.5
ORIGIN: https://github.com/Homebrew/homebrew
HEAD: 1f9845a90a8608386e2dfae5a3148f9af6ea6f16
Last commit: 21 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit broadwell
OS X: 10.11.3-x86_64
Xcode: N/A
Error: undefined method `/' for nil:NilClass
Please report this bug:
    https://git.io/brew-troubleshooting
/usr/local/Library/Homebrew/os/mac/xcode.rb:82:in `uncached_version'
/usr/local/Library/Homebrew/os/mac/xcode.rb:71:in `version'
/usr/local/Library/Homebrew/cmd/config.rb:43:in `clt'
/usr/local/Library/Homebrew/cmd/config.rb:154:in `dump_verbose_config'
/usr/local/Library/Homebrew/cmd/config.rb:7:in `config'
/usr/local/Library/brew.rb:83:in `<main>'
```

**Comment by tdsmith** [comment]
Fixed for now by c6183aa; thanks for reporting.
